src/Stratis.Bitcoin.Features.Miner/PosMinting.cs

```diff
@@ -509,67 +515,77 @@ public void StopStake()
-                this.rpcGetStakingInfoModel.CurrentBlockSize = block.GetSerializedSize();
-                this.rpcGetStakingInfoModel.CurrentBlockTx = block.Transactions.Count();
+                this.rpcGetStakingInfoModel.CurrentBlockSize = posBlock.GetSerializedSize();
+                this.rpcGetStakingInfoModel.CurrentBlockTx = posBlock.Transactions.Count();
                 this.rpcGetStakingInfoModel.PooledTx = await this.mempoolLock.ReadAsync(() => this.mempool.MapTx.Count).ConfigureAwait(false);
                 this.rpcGetStakingInfoModel.Difficulty = this.GetDifficulty(chainTip);
                 this.rpcGetStakingInfoModel.NetStakeWeight = this.networkWeight;
 
                 // Trying to create coinstake that satisfies the difficulty target, put it into a block and sign the block.
-                if (await this.StakeAndSignBlockAsync(utxoStakeDescriptions, block, chainTip, blockTemplate.TotalFee, coinstakeTimestamp).ConfigureAwait(false))
+                if (await this.StakeAndSignBlockAsync(utxoStakeDescriptions, posBlock, chainTip, blockTemplate.TotalFee, coinstakeTimestamp).ConfigureAwait(false))
                 {
                     this.logger.LogTrace("New POS block created and signed successfully.");
-                    this.CheckStake(block, chainTip);
+                    this.CheckStake(posBlock, chainTip);
 
                     blockTemplate = null;
                 }
                 else
                 {
-                    this.logger.LogTrace("{0} failed, waiting {1} ms for next round...", nameof(this.StakeAndSignBlockAsync), this.minerSleep);
+                    this.logger.LogTrace("{0} failed to create POS block, waiting {1} ms for next round...", nameof(this.StakeAndSignBlockAsync), this.minerSleep);
                     await Task.Delay(TimeSpan.FromMilliseconds(this.minerSleep), this.stakeCancellationTokenSource.Token).ConfigureAwait(false);
                 }
             }
         }
 
+        private async Task<List<UtxoStakeDescription>> GetUtxoStakeDescriptions(WalletSecret walletSecret, CancellationToken ct)
+        {
+            var utxoStakeDescriptions = new List<UtxoStakeDescription>();
+            IEnumerable<UnspentOutputReference> spendableTransactions = this.walletManager
+                .GetSpendableTransactionsInWallet(walletSecret.WalletName, 1)
+                .Where(t => t != null).ToList();
+
+            //todo 
+            FetchCoinsResponse fetchedCoinSet = await this.coinView.FetchCoinsAsync(spendableTransactions.Select(t => t.Transaction.Id).ToArray(), ct).ConfigureAwait(false);
+            if (ct.IsCancellationRequested) return utxoStakeDescriptions;
+
+            foreach (UnspentOutputReference outputReference in spendableTransactions)
+            {
+                UnspentOutputs coinSet = fetchedCoinSet.UnspentOutputs
+                    .FirstOrDefault(f => f.TransactionId == outputReference.Transaction.Id);
```

Comment:
looks like you have removed `?` here, are you sure it can't be null?

Reply:
also the original line was not that much long, you didn't need to split it

Reply:
Oh F** yes I think it can actually be null you are right, just saw a way in DBreeze coinview implementation, thanks